collector/src/main/java/io/prometheus/jmx/JmxCollector.java

```diff
@@ -244,22 +246,28 @@ private Config loadConfig(Map<String, Object> yamlConfig) throws MalformedObject
 
     }
 
+    private static final ConcurrentHashMap<String, String> snakeCache = new ConcurrentHashMap<String, String>();
```

Comment:
This can leak memory in certain (unwise) configurations, so we shouldn't do this.

Reply:
In fact it *could* be if MBeans attributes names were changed at each scrape. This won't be the case : the list of "snake cased" attributes will be stable over time I guess.
What's true is the map would hold lots of Entries. But as they're Strings, the memory footprint is not heavy.
Again, may be this can be disabled via configuration if necessary.

Reply:
It's probably best not to make that assumption, there's a lot of weird stuff out there. Anything like this should work correctly out of the box, without any configuration options.